geode-lucene/src/main/java/org/apache/geode/cache/lucene/FlatFormatSerializer.java

```diff
@@ -140,13 +140,17 @@ private Object getFieldValue(Object value, String fieldName) {
           && SerializerUtil.supportedPrimitiveTypes().contains(clazz)) {
         return value;
       }
-      try {
-        Field field = clazz.getDeclaredField(fieldName);
-        field.setAccessible(true);
-        return field.get(value);
-      } catch (Exception e) {
-        return null;
-      }
+
+      do {
+        try {
+          Field field = clazz.getDeclaredField(fieldName);
+          field.setAccessible(true);
+          return field.get(value);
+        } catch (Exception e) {
+          clazz = clazz.getSuperclass();
```

Comment:
Seems like it would be better to specifically check to see if the field is missing, rather than catch a blanket "Exception" here.

Reply:
it's old behavior. I kept it.